Topics/Tech_Stacks/ORM.md

```diff
@@ -0,0 +1,133 @@
+# ORM (Object Relation Mapper)
+
+## Table of Content
+### [Introductions](#introductions-1)
+### [What is an ORM](#what-is-an-orm-1)
+### [Example](#example-1)
+### [Should I use one?](#should-i-use-one-1)
+### [Advantages](#advantages-1)
+### [Disadvantages](#disadvantages-1)
+### [Conclusion](#conclusion-1)
+
+## Introductions:
+Most applications that you will build, in fact most applications that exist today, require some sort of data permanence. Meaning that data is stored somewhere, and is not lost when a site/app is reloaded. We employ databases like MYSQL, PostgreSQL, and MongoDB to ensure that our data can we stored and we can safely interact with it. But how do we do that? There are many database drivers that can allow us to write queries as lines of code directly from within our javascript or python (or whatever language you prefer) scripts. On the other hand, you may consider using an ORM to write your database code!
+
+
+## What is an ORM:
+
+An ORM (Object Relation Mapper) is a software tool that allows for the translation of an entity between its representation in a database (A relation) and object-oriented programming languages (An Object).
```

Comment:
Since ORM seems like it is a tool for relational databases, like SQL, it might be worth mentioning if it can also be applied to non-relational databases, such as MongoDB, or if MongoDB uses a different mechanism. As far as I know, I believe that Mongo uses ODMs.